esdt-safe/src/lib.rs

```diff
@@ -218,29 +218,27 @@ pub trait EsdtSafe:
         }
     }
 
-    // endpoints
-
-    /// Create an MultiversX -> Ethereum transaction. Only fungible tokens are accepted.
-    ///
-    /// Every transfer will have a part of the tokens subtracted as fees.
-    /// The fee amount depends on the global eth_tx_gas_limit
-    /// and the current GWEI price, respective to the bridged token
-    ///
-    /// fee_amount = price_per_gas_unit * eth_tx_gas_limit
-    #[payable("*")]
-    #[endpoint(createTransaction)]
-    fn create_transaction(&self, to: EthAddress<Self::Api>) {
+    fn create_transaction_common(
+        &self,
+        to: EthAddress<Self::Api>,
+    ) -> (
+        u64,
```

Comment:
I would create a type that contains these items so it can be easily readable.
At least for the signature it would be really helpful.
I can help with that.